### PR TITLE
Enable TTL compaction filter by default and deprecate/remove some other TTL API

### PR DESCRIPTION
**Pull request by azagrebin**
## What is the purpose of the change

Addressing [FLINK-15605](https://issues.apache.org/jira/browse/FLINK-15605), [FLINK-15606](https://issues.apache.org/jira/browse/FLINK-15606), [FLINK-15506](https://issues.apache.org/jira/browse/FLINK-15506).

## Brief change log

  - remove `StateTtlConfig.TimeCharacteristic`
  - deprecate `StateTtlConfig.cleanupInBackground`
  - set `RocksDBOptions.TTL_COMPACT_FILTER_ENABLED.defaultValue` to `true` and deprecate it

## Verifying this change

trivial change

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (yes)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs / JavaDocs)

### CODE REVIEW

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java [hunk 1]
```diff
@@ -59,10 +59,15 @@
...
+	 * This determines if compaction filter to cleanup state with TTL is enabled.
+	 *
+	 * @deprecated the option will be removed in future and have no effect.
+	 */
+	@Deprecated
 	public static final ConfigOption<Boolean> TTL_COMPACT_FILTER_ENABLED = ConfigOptions
 		.key("state.backend.rocksdb.ttl.compaction.filter.enabled")
-		.defaultValue(false)
+		.defaultValue(true)
 		.withDescription("This determines if compaction filter to cleanup state with TTL is enabled for backend." +
```
Comment (tillrohrmann) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java]: Whitespace is missing after `.`

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java [hunk 2]
```diff
@@ -59,10 +59,15 @@
 		.defaultValue(1)
 		.withDescription("The number of threads (per stateful operator) used to transfer (download and upload) files in RocksDBStateBackend.");
 
-	/** This determines if compaction filter to cleanup state with TTL is enabled. */
+	/**
+	 * This determines if compaction filter to cleanup state with TTL is enabled.
+	 *
+	 * @deprecated the option will be removed in future and have no effect.
```
Comment (tillrohrmann) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java]:
```suggestion
	 * @deprecated the option will be removed in the future and should only be used if experiencing serious performance degradations.
```

Comment (zentol) [reply flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java]: `should only be used >when< experiencing serious ...`

Comment (tillrohrmann) [reply flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java]: Good point. Will change it while merging the PR.

**Comment by tillrohrmann** [comment]
@flinkbot run travis

**Comment by tillrohrmann** [comment]
@flinkbot run travis

**Comment by azagrebin** [comment]
Thanks for the review @tillrohrmann @zentol 
merged into master by 4a66f107b0e95d4729b777e9f289c2d42f95de28
merged into 1.10 by 72a0158daeebbd821aa5d68a212f83648f4b1572

**Pull request closed without merging**